gitlab/v4/objects/invitations.py

```diff
@@ -0,0 +1,91 @@
+from typing import Any, cast, Union
+
+from gitlab.base import RESTManager, RESTObject
+from gitlab.exceptions import GitlabInvitationError
+from gitlab.mixins import CRUDMixin, ObjectDeleteMixin, SaveMixin
+from gitlab.types import CommaSeparatedListAttribute, RequiredOptional
+
+__all__ = [
+    "ProjectInvitation",
+    "ProjectInvitationManager",
+    "GroupInvitation",
+    "GroupInvitationManager",
+]
+
+
+class InvitationMixin(CRUDMixin):
+    def create(self, *args: Any, **kwargs: Any) -> RESTObject:
+        invitation = super().create(*args, **kwargs)
+
+        if invitation.status == "error":
+            raise GitlabInvitationError(invitation.message)
+
+        return invitation
+
+
+class ProjectInvitation(SaveMixin, ObjectDeleteMixin, RESTObject):
+    _id_attr = "email"
+
+
+class ProjectInvitationManager(InvitationMixin, RESTManager):
+    _path = "/projects/{project_id}/invitations"
+    _obj_cls = ProjectInvitation
+    _from_parent_attrs = {"project_id": "id"}
+    _create_attrs = RequiredOptional(
+        required=("access_level",),
+        optional=(
+            "expires_at",
+            "invite_source",
+            "tasks_to_be_done",
+            "tasks_project_id",
+        ),
+        exclusive=("email", "user_id"),
+    )
+    _update_attrs = RequiredOptional(
+        required=("access_level",),
```

Comment:
According to the docs `access_level` is not required when updating.

Reply:
Oh thanks! copy/paste gone wrong :grin: